src/directory_listing.rs

```diff
@@ -589,3 +588,70 @@ fn parse_last_modified(modified: SystemTime) -> Result<DateTime<Local>> {
         )),
     }
 }
+
+/// Formats the file size in bytes to a human-readable string
+fn format_file_size(size: u64) -> String {
+    let units = ["B", "KiB", "MiB", "GiB", "TiB", "PiB"];
+    let mut size_tmp = size;
+
+    for i in 0..units.len() - 1 {
+        if size_tmp < 1024 {
+            // return the size with the current unit
+            return format!("{} {}", size_tmp, units[i]);
```

Comment:
This case doesn’t need to be inside the loop – you can have bytes as a special case before it.